### PR TITLE
feat: Deliver MultiQC intermediate files

### PR DESCRIPTION
**Pull request by ivadym**
## Description

Add MultiQC intermediates files to the deliverables `.hk` file.

#### Added
- MultiQC intermediate files to deliverables

#### Changed

- `delivery.py`, removing unused arguments 

## Documentation

- [x] N/A
- [ ] Updated Balsamic documentation to reflect the changes as needed for this PR.
    - [Document Name]

### Feature Tests

- [ ] N/A
- [x] Test generation of the `.hk` file
    Screenshot 2024-02-05 at 14 51 31
- [x] Test MultiQC files included in the deliverables `.hk` file with unique tags
    Screenshot 2024-02-05 at 14 52 54
- [x] Test that different workflow generate the intermediate MultiQC expected files

### Pipeline Integrity Tests

- **Report deliver (generation of the `.hk` file)**
    - [ ] N/A
    - [x] Verified
- **TGA T/O Workflow**
    - [x] N/A
    - [ ] Verified
- **TGA T/N Workflow**
    - [x] N/A
    - [ ] Verified
- **UMI T/O Workflow**
    - [x] N/A
    - [ ] Verified
- **UMI T/N Workflow**
    - [x] N/A
    - [ ] Verified
- **WGS T/O Workflow**
    - [x] N/A
    - [ ] Verified
- **WGS T/N Workflow**
    - [x] N/A
    - [ ] Verified
- **QC Workflow**
    - [x] N/A
    - [ ] Verified
- **PON Workflow**
    - [x] N/A
    - [ ] Verified

## Clinical Genomics Stockholm

<details>
<summary></summary>

### Documentation

- **Atlas documentation**
    - [x] N/A
    - [ ] Updated: [Link]
- **Web portal for Clinical Genomics**
    - [x] N/A
    - [ ] Updated: [Link]

### User Changes

- [x] N/A
- [ ] This PR affects the output files or results.
    - [ ] User feedback is considered unnecessary because [Justification].
    - [ ] Affected users have been included in the development process and given a chance to provide feedback.

### Infrastructure Changes

- **Stored files in Housekeeper**
    - [ ] N/A
    - [x] Updated: https://github.com/Clinical-Genomics/hermes/pull/95
- **CG (CLI and delivered/uploaded files)**
    - [x] N/A
    - [ ] Updated: [Link]
- **Servers (configuration files on Hasta)**
    - [x] N/A
    - [ ] Updated: [Link]
- **Scout interface**
    - [x] N/A
    - [ ] Updated: [Link]

### Integration Tests

- [ ] N/A
- [x] Test MultiQC intermediate files stored in HK
    - Tested in https://github.com/Clinical-Genomics/hermes/pull/95

</details>

## Checklist

> [!IMPORTANT]  
> Ensure that all checkboxes below are ticked before merging.

### For Developers

- **PR Description**
    - [x] Provided a comprehensive description of the PR.
    - [x] Linked relevant user stories or issues to the PR.
- **Documentation**
    - [x] Verified and updated documentation if necessary.
- **Tests**
    - [x] Described and tested the functionality addressed in the PR.
    - [x] Ensured integration of the new code with existing workflows.
    - [x] Confirmed that meaningful unit tests were added for the changes introduced.
    - [x] Checked that the PR has successfully passed all relevant code smells and coverage checks.
- **Review**
    - [x] Addressed and resolved all the feedback provided during the code review process.
    - [x] Obtained final approval from designated reviewers.

### For Reviewers

- **Code**
    - [x] Code implements the intended features or fixes the reported issue.
    - [x] Code follows the project's coding standards and style guide.
- **Documentation**
    - [x] Pipeline changes are well-documented in the CHANGELOG and relevant documentation.
- **Tests**
    - [ ] The author provided a description of their manual testing, including consideration of edge cases and boundary
      conditions where applicable, with satisfactory results.
- **Review**
    - [ ] Confirmed that the developer has addressed all the comments during the code review.

### CODE REVIEW

File: BALSAMIC/utils/delivery.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+from BALSAMIC.constants.constants import FileType
+
+LOG = logging.getLogger(__name__)
+
+
+def get_multiqc_deliverables(case_id: str, multiqc_dir: Path) -> List[Dict[str, Any]]:
+    """Return a list of MultiQC deliverable files from a directory."""
+    multiqc_deliverables: List[Dict[str, Any]] = []
+    json_files: Generator[Path, None, None] = multiqc_dir.glob(f"*.{FileType.JSON}")
+    for file in json_files:
```
Comment (mathiasbio) [inline BALSAMIC/utils/delivery.py:17]: I just have to ask, since only results from the tests of one case has been reported in this PR, have you considered if there are any edge-case scenarios with unique multiqc_json reports created from the other workflows?

**Review by mathiasbio — APPROVED** [review]
This looks great 🌟 Looks like you also took the opportunity to do some refactoring. Feels like this deliver.py is easier to read now! I just added a question, but even if you haven't tested it explicitly I trust in the logic of your code. Just wanted to make sure that it has been considered : )

File: BALSAMIC/utils/delivery.py (hunk 1 above)
Comment (ivadym) [reply BALSAMIC/utils/delivery.py:17]: Good point! I actually did check which files each workflow generated. I compared them to see if there were any major differences, but it looks like the only changes are in the Picard WGS metrics and Somalier for Tumor/Normal samples. In any case, here we'll be including all the json files in the deliverables and then filter further in the Hermes PR :)

https://docs.google.com/spreadsheets/d/1OS5UVIFcSVdcFNCAbJZehUn15OB6_UtVTr6FSaccSkU/edit#gid=606705712
